pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -81,12 +82,12 @@ func dataDirRelativeHostPath(monName string) string {
 }
 
 // LoadClusterInfo constructs or loads a clusterinfo and returns it along with the maxMonID
-func LoadClusterInfo(ctx *clusterd.Context, context context.Context, namespace string) (*cephclient.ClusterInfo, int, *Mapping, error) {
-	return CreateOrLoadClusterInfo(ctx, context, namespace, nil)
+func LoadClusterInfo(ctx *clusterd.Context, context context.Context, namespace string, clusterSpec cephv1.ClusterSpec) (*cephclient.ClusterInfo, int, *Mapping, error) {
+	return CreateOrLoadClusterInfo(ctx, context, namespace, nil, clusterSpec)
 }
 
 // CreateOrLoadClusterInfo constructs or loads a clusterinfo and returns it along with the maxMonID
-func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.Context, namespace string, ownerInfo *k8sutil.OwnerInfo) (*cephclient.ClusterInfo, int, *Mapping, error) {
+func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.Context, namespace string, ownerInfo *k8sutil.OwnerInfo, clusterSpec cephv1.ClusterSpec) (*cephclient.ClusterInfo, int, *Mapping, error) {
```

Comment:
Ok I know you brought up this question sooner, but I'm going to suggest a simpler approach so we can avoid changing the parameters to this method that affects so many places. Really, we only want to update the annotations in the cluster reconcile instead of every time this method is called, which is frequently. Instead of updating the annotations from this method, let's update it from the cluster controller in one place after it calls this method. Then we don't need to change all the callers to this method. Thanks!